Adjusting PCA computation and use of halo hits

artlbv opened the pull request:
In order to improve the PCA logic and also take into account the halo issue in #43, the following changes are proposed:

- Changing recomputePCA logic:
  - New separate function to recompute PCA
  - PCA is recomputed before any other shape variables (sigmas)
- Flag allowing to use also Halo hits within 5cm cylinder around axis for PCA/variables
- Added energy/pt of multicluster computed from hits within the cylinder

The improvements in terms of performance are mostly related to the use of the previously ignored halo hits. The use of halo hits in proximity to the cluster axis recovers the hits that were accidentally labeled as halo.

Effect of changes (blue histogram):
- Outliers in sigmaVV/UU (radial/perpendicular direction) are reduced
- SigmaUU/VV shift slightly, but distribution gets narrower

![image](https://user-images.githubusercontent.com/4972492/30222602-927af954-94c7-11e7-93c9-f9896fed3b23.png)

clelange commented:
Hi @artlbv - looks good to me, and Florian and you are the only users, so I assume you're both happy with it.
Two requests: could you please
- adjust the NTUP template in https://github.com/CMS-HGCAL/reco-prodtools accordingly adding the new flag
- undo commit https://github.com/CMS-HGCAL/reco-ntuples/pull/48/commits/dcc460c896ded68c691293affe3c29673452c17b unless you have a good reason why the clustering should not be rerun in the example.

artlbv commented:
Hi @clelange, ok, I will make a PR with the updated to reco-prodtools then.
Concerning reruning the clustering – didn't you mention we should not rerun it on the new samples? Or is it just that this flag should remain True in any case?

clelange commented:
@artlbv Since the config just serves as an example, I would keep it set to true, but then again I have no strong opinion, it's just more convenient for clustering debugging. The resulting ntuple should be the same anyway.

artlbv commented:
1. Reverted dcc460c (rerunning clustering by default)
2. Updated config in ntup template, see PR https://github.com/CMS-HGCAL/reco-prodtools/pull/34 
Ready to merge.
